### ISSUE TITLE
Uploading new photo to existing observation does not complete progress

### ISSUE THREAD

**Issue by kueda** [inaturalist/iNaturalistReactNative#918]
**Describe the bug**
When you upload a new photo for an existing observation, the circular progress indicator on the observation never completes its progress.

**To Reproduce**
1. Add a new observation with a photo and upload it
2. Tap on that observation to get to ObsDetail
3. Tap on the pencil button in the upper right get to ObsEdit
4. Tap the "+" button in the EVIDENCE section and choose to import another photo
5. Tap the "SAVE " button to save the observation
6. Return to My Observations. The toolbar should show the sync button with the "Upload 1 observation" label and the observation list item should show its own upload button
7. Tap "Upload 1 observation" in the toolbar
8. The upload should complete and the toolbar UI should return to its default state, but the circular progress indicator on the observation list item remains in a state of incomplete progress
9. Tap on another observation to get to ObsDetail and then return to MyObs
10. The circular progress indicator has gone away and we're showing obs status instead

**Expected behavior**
The circular progress indicator should complete whether you are uploading an observation or just the photos.

**Context (please complete the following information):**
 - Device: iPad Air (5th gen)
 - OS: iPadOS 16.6.1
 - App version: [v0.19.0+67](https://github.com/inaturalist/iNaturalistReactNative/releases/tag/v0.19.0%2B67)

Originally reported from iPhone 13 Pro, iOS 17.0.3 with this report:

> Make an obs and sync it. Then, edit it and add a photo to it and start syncing again. The sync animation stops at this point (see below), but the spinner keeps going ad infinitum. Go to another screen then back to the my obs screen and you’ll see the obs has synced correctly. Maybe related to https://github.com/inaturalist/iNaturalistReactNative/issues/905

**Additional context**
I noticed a few other hiccups when while trying to reproduce this, particularly when tapping "UPLOAD NOW" instead of saving and uploading all obs from the toolbar.

**Comment by kueda** [comment]
Currently blocked by [#993](https://github.com/inaturalist/iNaturalistReactNative/issues/993)

**Comment by joergmlpts** [comment]
I looked into this and noticed the new issue #993 as well. In a debug build on Android #993 looks like this:

![Screenshot_20231230-144138](https://github.com/inaturalist/iNaturalistReactNative/assets/66563749/6b0410c7-64fb-4d7c-9950-a335dd9383d4)

That happens in `Observation.appendObsPhotos()` where in
 
```
   const updatedObs = currentObservation;
   updatedObs.observationPhotos = [...currentObservationPhotos, ...obsPhotos];
```

currentObservation cannot be modified. A change that appears to work for me is `const updatedObs = currentObservation.toJSON( )`.

With this fix for #993 in place, the initial issue can be reproduced:

![Screenshot_20231230-210445](https://github.com/inaturalist/iNaturalistReactNative/assets/66563749/18aa298d-f678-4d96-a134-29a726788a75)

This happens in uploadEvidence when `currentEvidence?.photo` is `null`. It is `null` for the photos that have been uploaded before.
